modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/spi/display/PermissionPropagationCTDisplayRenderer.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.change.tracking.web.internal.spi.display;
+
+import com.liferay.change.tracking.spi.display.BaseCTDisplayRenderer;
+import com.liferay.change.tracking.spi.display.CTDisplayRenderer;
+import com.liferay.document.library.kernel.service.DLFolderLocalService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.model.ClassName;
+import com.liferay.portal.kernel.model.PermissionPropagationEntry;
+import com.liferay.portal.kernel.security.permission.ResourceActions;
+import com.liferay.portal.kernel.security.permission.ResourceActionsUtil;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Locale;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Quan Huynh
+ */
+@Component(service = CTDisplayRenderer.class)
+public class PermissionPropagationCTDisplayRenderer
+	extends BaseCTDisplayRenderer<PermissionPropagationEntry> {
+
+	@Override
+	public Class<PermissionPropagationEntry> getModelClass() {
+		return PermissionPropagationEntry.class;
+	}
+
+	@Override
+	public String getTitle(
+			Locale locale,
+			PermissionPropagationEntry permissionPropagationEntry)
+		throws PortalException {
+
+		ClassName className = _classNameLocalService.fetchClassName(
+			permissionPropagationEntry.getClassNameId());
+
+		if (className == null) {
+			return null;
+		}
+
+		String[] classNames = StringUtil.split(
+			className.getClassName(),
+			ResourceActionsUtil.getCompositeModelNameSeparator());
```

Comment:
What's the reason to do this?

Reply:
Shouldn't use _Util_ class. Please use **@Reference** to inject.

Reply:
Fixed it. Thanks!